scripts/jest/jest-cli.js

```diff
@@ -317,7 +317,17 @@ function main() {
   }
 
   // Run Jest.
-  spawn('node', args, {stdio: 'inherit', env: {...envars, ...process.env}});
+  const process = spawn('node', args, {
+    stdio: 'inherit',
+    env: {...envars, ...process.env},
+  });
+  // Ensure we close our process when we get a failure case.
+  process.on('close', code => {
+    // If we get a failure case then stop the main process.
```

Comment:
This reads as though it will stop on the first failure, but it doesn't stop until Jest completes:

```suggestion
    // Forward the exit code from the Jest process.
```